compiler/z/codegen/OMRCodeGenerator.hpp

```diff
@@ -237,7 +237,7 @@ class OMR_EXTENSIBLE CodeGenerator : public OMR::CodeGenerator
     */
    bool materializesLargeConstants() { return true; }
    bool shouldValueBeInACommonedNode(int64_t value);
-   int64_t getLargestNegConstThatMustBeMaterialized() {return ((-1ll) << 31) - 1;}   // min 32bit signed integer minus 1
+   int64_t getLargestNegConstThatMustBeMaterialized() { return ((int64_t)~(~((uint64_t)0) << 31)) - 1;}
```

Comment:
This doesn't yield the same value (we don't want two uses of `~`); and the comment should stay:
```c
   int64_t getLargestNegConstThatMustBeMaterialized() { return (int64_t)(((~(uint64_t)0) << 31) - 1); } // min 32-bit signed integer minus 1
```